src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -775,32 +775,6 @@ private static boolean isUserDefinedArrangementOfThis(AbstractFrame currentFrame
         return userDefinedArrangementOfThis;
     }
 
-    /**
-     * Returns the token which ends the code block.
-     *
-     * @param blockNameIdent block name identifier.
-     * @param blockStartToken token which starts the block.
-     * @return the token which ends the code block.
-     */
-    private static DetailAST getBlockEndToken(DetailAST blockNameIdent, DetailAST blockStartToken) {
-        DetailAST blockEndToken = null;
-        final DetailAST blockNameIdentParent = blockNameIdent.getParent();
-        if (blockNameIdentParent.getType() == TokenTypes.CASE_GROUP) {
-            blockEndToken = blockNameIdentParent.getNextSibling();
-        }
-        else {
-            final Set<DetailAST> rcurlyTokens = getAllTokensOfType(blockNameIdent,
-                    TokenTypes.RCURLY);
-            for (DetailAST currentRcurly : rcurlyTokens) {
-                final DetailAST parent = currentRcurly.getParent();
-                if (TokenUtil.areOnSameLine(blockStartToken, parent)) {
-                    blockEndToken = currentRcurly;
-                }
-            }
-        }
-        return blockEndToken;
-    }
-
```

Comment:
I went over the cases, this should be fine. The block end token is the last child of the block start token. Though variables in switches can also be globally (by globally I mean throughout the switch statement) scoped, this shouldn't cause a problem as the variable would have to be initialized separately in every switch case. Diff report is also clean.

For switches, the last child will be the last statement in that block.